hack/boilerplate/boilerplate.go.txt

```diff
@@ -0,0 +1,16 @@
+/*
+Copyright YEAR The Karmada Authors.
```

Comment:
A question:
I guess this `YEAR` will be replaced with the year when generating the auto-generated files, right?

Reply:
Does this will block the first PR of next year? 
I mean block by [hack/verify-codegen.sh](https://github.com/karmada-io/karmada/blob/master/hack/verify-codegen.sh).

Reply:
Good to know. 
+1
I didn't know Kubernetes now using this boilerplate.

Reply:
thanks for your good review.